packages/menu/src/Menu/Menu.tsx

```diff
@@ -110,10 +113,20 @@ export const Menu = React.forwardRef<HTMLDivElement, MenuProps>(function Menu(
     }
   }, [setOpen, shouldClose]);
 
+  const [, updateState] = React.useState();
+  // Used to trigger a state update when the current subMenu changes since the current subMenu is stored in a ref to avoid extra rerenders on initial load.
+  // @ts-expect-error - updateState should have a value
+  const updateCurrentSubMenu = React.useCallback(() => updateState({}), []);
+
   const triggerRef = useRef<HTMLElement>(null);
+  // This hook causes a second re-render on initial load. `useAvailableSpace` uses `useViewportSize` internally, which has internal state that causes re-renders.
```

Comment:
I'm not sure of the best way to do this; I was thinking this inside `useViewPortSize`:
```
const memoizedViewportSize = useMemo(() => viewportSize, [viewportSize]);
return memoizedViewportSize;
```

Does it make sense to do this since the state would only change once (if there is no resizing) when the viewport is available in the useEffect hook?

Reply:
That's true, it would still rerender once even if we memoized. 
I wonder if setting a better default would fix it though:
```ts
  const [viewportSize, setViewportUpdateVal] = useState<ViewportSize>(
    getViewportSize()
  );
```
(and handling the case where window is undefined within `getViewportSize`)

Reply:
I think I should create a ticket for this since this hook is used in other components.

Reply:
https://jira.mongodb.org/browse/LG-3327